pkg/projects/resource_project_membership.go

```diff
@@ -196,26 +197,16 @@ var updateMember = func(projectKey string, membershipType string, member Member,
 	return err
 }
 
-var deleteMembers = func(projectKey string, membershipType string, members []Member, m interface{}) error {
+var deleteMembers = func(projectKey string, membershipType string, members []Member, m interface{}, g *errgroup.Group) {
 	log.Println("[DEBUG] deleteMembers")
 
-	if membershipType != usersMembershipType && membershipType != groupssMembershipType {
-		return fmt.Errorf("Invalid membershipType: %s", membershipType)
-	}
-
-	var errs []error
 	for _, member := range members {
-		err := deleteMember(projectKey, membershipType, member, m)
-		if err != nil {
-			errs = append(errs, err)
-		}
-	}
+		projectKey, membershipType, member, m := projectKey, membershipType, member, m
 
-	if len(errs) > 0 {
-		return fmt.Errorf("failed to delete members from project: %v", errs)
+		g.Go(func() error {
```

Comment:
I wish this library took a func param of `func() (interface{}, error)` that way your end result would be 2 slices:

```go
[]interface{}, []error
```
corresponding to each call

Reply:
I agree. More like `Promise.allSettled()` in JS.